aten/src/ATen/core/Dict_inl.h

```diff
@@ -50,6 +50,8 @@ inline size_t DictKeyHash::operator()(const IValue& ivalue) const {
     return std::hash<bool>()(ivalue.toBool());
   } else if (ivalue.isTensor()) {
     return std::hash<TensorImpl*>()(ivalue.toTensor().unsafeGetTensorImpl());
+  } else if (ivalue.isDevice()) {
+    return std::hash<std::string>()(ivalue.toDevice().str());
```

Comment:
There already exists a specialization of `std::hash` for devices: https://github.com/pytorch/pytorch/blob/b9fc656cf26d60127bd695e4e5a7d27622f2563d/c10/core/Device.h#L137

Hence this could be simplified, and made more idiomatic and efficient, by doing:
```suggestion
    return std::hash<Device>()(ivalue.toDevice());
```

Reply:
Thanks for suggestion!